### ISSUE TITLE
punch silently adds and commits unrelated untracked files to git

### ISSUE THREAD

**Issue by kshpytsya** [lgiordani/punch#29]
Punch version 1.4.2
os: Gentoo Linux
git version 2.16.2
```sh
$ git init
Initialized empty Git repository in /tmp/punch-bug/.git/
$ punch --init
```
...configure files and punch once...
```
$ cat punch_config.py
__config_version__ = 1

GLOBALS = {
    'serializer': '{{major}}.{{minor}}.{{patch}}',
}

FILES = ["VERSION"]

VERSION = ['major', 'minor', 'patch']

VCS = {
    'name': 'git',
    'commit_message': "Version updated from {{ current_version }} to {{ new_version }}",
}
$ cat punch_version.py
major = 0
minor = 2
patch = 0
$ cat VERSION
0.2.0
$ git status
On branch master
nothing to commit, working tree clean
$ touch a
$ git status
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        a

nothing added to commit but untracked files present (use "git add" to track)
$ punch -p minor
$ git status
On branch master
nothing to commit, working tree clean
$ git log -n1 -p
commit 7c975201be6ec6a070d4f11d64df6e296c9ffd33 (HEAD -> master, tag: 0.3.0)
Author: Kyrylo Shpytsya <kshpitsa@gmail.com>
Date:   Tue Feb 27 22:40:38 2018 +0200

    Version updated from 0.2.0 to 0.3.0

diff --git a/VERSION b/VERSION
index 0ea3a94..0d91a54 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-0.2.0
+0.3.0
diff --git a/a b/a
new file mode 100644
index 0000000..e69de29
diff --git a/punch_version.py b/punch_version.py
index 1085779..31aa2c8 100644
--- a/punch_version.py
+++ b/punch_version.py
@@ -1,3 +1,3 @@
 major = 0
-minor = 2
+minor = 3
 patch = 0
```
Observe the untracked file `a` being silently added and committed.

**Comment by kshpytsya** [comment]
P.S. You should really consider using `--porcelain` git flag, just as `bumpversion` does, instead of parsing human-readable git output.

**Comment by lgiordani** [comment]
Hey thanks for the issue and for the advice, I'll work on it!

**Comment by lgiordani** [comment]
Solved in version 1.4.3. Thanks for spotting it!
Thanks for the advice about `--porcelain`, I actually already started a branch with a refactoring of that part, so good to receive positive pushed towards that direction =)
